modules/tensor_mechanics/src/kernels/GradScalar.C

```diff
@@ -0,0 +1,53 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "GradScalar.h"
+#include "Function.h"
+
+template<>
+InputParameters validParams<GradScalar>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription("Adds -coefficient*scalar*grad_test[component]");
+  params.addParam<FunctionName>("coefficient", 1, "Coefficient.  May be a Real, or a Function");
+  params.addRequiredCoupledVar("scalar", "Scalar.  This kernel adds -coefficient*scalar*grad_test[component]");
+  params.addRequiredParam<unsigned int>("component", "The gradient direction (0 for x, 1 for y and 2 for z)");
+  return params;
+}
+
+GradScalar::GradScalar(const std::string & name, InputParameters parameters) :
+  Kernel(name, parameters),
+  _coefficient(getFunction("coefficient")),
+  _scalar(coupledValue("scalar")),
+  _scalar_var_num(coupled("scalar")),
+  _component(getParam<unsigned int>("component"))
+{
+  if (_component >= _mesh.dimension())
+    mooseError("GradScalar: component should not be greater than the mesh dimension\n");
+}
+
+Real
+GradScalar::computeQpResidual()
+{
+  return -_coefficient.value(_t, _q_point[_qp])*_scalar[_qp]*_grad_test[_i][_qp](_component);
+}
+
+
+Real
+GradScalar::computeQpJacobian()
+{
+  if (_var.number() != _scalar_var_num)
```

Comment:
Also: For the same reason, I'm not a huge fan of the name as it seems to imply that it's _for_ Scalar variables... instead of just coupled to them.

Is there a better (maybe more physical) name you could use?